specification/cognitiveservices/OpenAI.Inference/models/completions/chat_messages.tsp

```diff
@@ -0,0 +1,202 @@
+import "./tools.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+#suppress "@azure-tools/typespec-autorest/union-unsupported" "External API shape is defined in OpenAPI 3.0 as oneOf."
+alias ChatMessageContent = string | ChatMessageContentItem[];
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("An abstract representation of a structured content item within a chat message.")
+@discriminator("type")
+model ChatMessageContentItem {
+  @doc("The discriminated object type.")
+  type: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A structured chat content item containing plain text.")
+model ChatMessageTextContentItem extends ChatMessageContentItem {
+  @doc("The discriminated object type: always 'text' for this type.")
+  type: "text";
+
+  @doc("The content of the message.")
+  text: string;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A representation of the possible image detail levels for image-based chat completions message content.")
+enum ChatMessageImageDetailLevel {
+  @doc("Specifies that the model should determine which detail level to apply using heuristics like image size.")
+  auto: "auto",
+
+  @doc("""
+  Specifies that image evaluation should be constrained to the 'low-res' model that may be faster and consume fewer
+  tokens but may also be less accurate for highly detailed images.
+  """)
+  low: "low",
+
+  @doc("""
+  Specifies that image evaluation should enable the 'high-res' model that may be more accurate for highly detailed
+  images but may also be slower and consume more tokens.
+  """)
+  high: "high",
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("A structured chat content item containing an image reference.")
+model ChatMessageImageContentItem extends ChatMessageContentItem {
+  @doc("The discriminated object type: always 'image_url' for this type.")
+  type: "image_url";
+
+  @doc("An internet location, which must be accessible to the model,from which the image may be retrieved.")
+  url: url;
+
+  @doc("""
+  The evaluation quality setting to use, which controls relative prioritization of speed, token consumption, and
+  accuracy.
+  """)
+  detail?: ChatMessageImageDetailLevel;
+}
+
+@discriminator("role")
+@added(ServiceApiVersions.v2023_05_15)
+@doc("An abstract representation of a chat message as provided in a request.")
+model ChatRequestMessage {
+  @doc("The chat role associated with this message.")
+  role: ChatRole;
+}
+
+@added(ServiceApiVersions.v2023_05_15)
+@doc("""
+A request chat message containing system instructions that influence how the model will generate a chat completions
+response.
+""")
+model ChatRequestSystemMessage extends ChatRequestMessage {
+  @doc("The chat role associated with this message, which is always 'system' for system messages.")
+  role: ChatRole.system;
+
+  @doc("The contents of the system message.")
+  content: string;
+
+  @doc("An optional name for the participant.")
+  name?: string;
```

Comment:
This appears to be a pasting mistake. I don't see any `name` optional property in the yaml file. (looking at `chatCompletionRequestMessageSystem`, FYI)